src/hp/magicsock/conn.rs

```diff
@@ -1966,6 +1962,20 @@ impl Actor {
         match dm {
             disco::Message::Ping(ping) => {
                 inc!(MagicsockMetrics::RecvDiscoPing);
+                // if we get here we got a valid ping from an unknown sender
+                // so insert an endpoint for them
+                if unknown_sender {
+                    self.peer_map.insert_endpoint(EndpointOptions {
+                        conn_sender: self.conn.actor_sender.clone(),
+                        conn_public_key: self.conn.public_key.clone(),
+                        public_key: Some(sender.clone()),
+                        derp_addr: if derp_node_src.is_some() {
```

Comment:
this check should be `if is_derp` (I think)

Reply:
isn't that the same? But it looks nicer, so will change.